lib/pubsubstub/stream.rb

```diff
@@ -0,0 +1,26 @@
+class Stream
+  def initialize(&callback)
+    @callback = callback
+    @closed = false
+  end
+
+  def close
+    return if closed?
+    @closed = true
+  end
+
+  def each(&front)
+    @front = front
+    @callback.call(self)
+    close unless @keep_open
```

Comment:
Yeah my bad. I started from Sinatra's Stream class and removed the things we didn't need. So yeah we always close.